mmeds/util.py

```diff
@@ -1000,55 +1002,121 @@ def make_pheniqs_config(reads_forward, reads_reverse, barcodes_forward, barcodes
+                         input_dir,
+                         output_dir,
+                         verbose,
+                         filename_template=fig.QIIME_FILENAME_TEMPLATE,
+                         sample_id_cats=fig.QIIME_SAMPLE_ID_CATS,
+                         forward_barcode_cats=fig.QIIME_FORWARD_BARCODE_CATS,
+                         reverse_barcode_cats=fig.QIIME_REVERSE_BARCODE_CATS):
+    """
+    Strip reads with errors from demultiplexed fastq files and write new file content
+    This has only been tested using input that has been demultiplexed using 'pheniqs'
+    =================================================================================
+    :num_allowed_errors: Maximum number of errors in barcode pairs to not be stripped
+    :mapping_file: File containing sample IDs and barcode pairs
+    :input_dir: Directory containing demultiplexed fastq files
+    :output_dir: Directory in which to store stripped fastq files
+    :verbose: Print information to stdout
+    :filename_template: String with formatting for sample id and (1 or 2) for forward/reverse
+    :sample_id_cats: tuple with sample id header text
+    :forward_barcode_cats: tuple with forward barcode header text
+    :reverse_barcode_cats: tuple with reverse barcode header text
+    """
+    # Read in mapping file
+    output_content = {}
+    map_df = pd.read_csv(Path(mapping_file), sep='\t', header=[0, 1], na_filter=False)
 
-                line = f.readline()
+    # Only three columns are needed
+    sample_ids = map_df[sample_id_cats[0]][sample_id_cats[1]]
+    forward_barcodes = map_df[forward_barcode_cats[0]][forward_barcode_cats[1]]
+    reverse_barcodes = map_df[reverse_barcode_cats[0]][reverse_barcode_cats[1]]
 
-            # Write output file to output_dir
-            p_out = Path(output_dir) / filename
-            p_out.touch()
-            p_out.write_bytes(gzip.compress(out.encode('utf-8')))
-            print(count, 'written', filename)
-            count += 1
+    # Hash data for easy access
+    map_hash = {key: (value1, value2) for (key, value1, value2) in zip(sample_ids, forward_barcodes, reverse_barcodes)}
+    verbose_template = '{} Writing {}'
+    count = 0
+
+    # Generate output for each sample's forward and reverse input files
+    for key in map_hash:
+        forward_filename = filename_template.format(key, 1)
+        reverse_filename = filename_template.format(key, 2)
+
+        # Create output files
+        forward_output = Path(output_dir) / forward_filename
+        reverse_output = Path(output_dir) / reverse_filename
+        forward_output.touch()
+        reverse_output.touch()
+
+        # Forward Reads
+        count += 1
+        Logger.debug(verbose_template.format(count, forward_filename))
+        if verbose:
+            print(verbose_template.format(count, forward_filename))
+        # Generate error-stripped content, compress to gzip format, and write output
+        forward_output.write_bytes(
```

Comment:
why not just forward_output.write() and then you can do away with all the utf8 encoding stuff in get_stripped_file_content like:
`content = b''`